app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -40,7 +40,8 @@ def manage_oauth_callback
       else
         # the certified email is already present in another account in Airesis
         # cannot update user email with one already taken
-        if user_info[:certified] && User.where(email: user_info[:email]).exists?
+        if user_info[:certified] &&
+           User.all_except(current_user).where(email: user_info[:email]).exists?

```

Comment:
forse il controllo può essere messo dentro a oauth_data_parser come metodo? che dici?
Poi potresti avere User.another_by_email(current_user.id, user_info[:email]).exists?


scope :another_by_email, ->(user_id, email) { where(email: email).where.not(id: user_id)}